src/main/java/graphs/GraphClass.java

```diff
@@ -43,6 +43,22 @@ public void addNode(GraphNode node) {
 
     public void addEdge(GraphEdge edge) {
         edges.add(edge);
+        if(! isDirected()) {
+            duplicateAsUndirectedEdge(edge);
+        }
+    }
+
+    private void duplicateAsUndirectedEdge(GraphEdge edge) {
+        edges.add(new GraphEdge(edge.getRightNode(), edge.getLeftNode(), edge.getWeight()));
+    }
+
+    public GraphClass duplicateUndirectedEdges() {
```

Comment:
what i am thinking now, is change the Graph constructors to take the isDirected flag, and then set that first in the constructor, and then say if(isDirected()) this.edges = new List(edges) else for(edge : edges) addEdge(edge)
this will either just put the edges into the graphs edge list, or if it's undirected, add each edge individually with the public addEdge() method which would duplicate the edge for both directions. then I can remove the public duplicateEdges() method, and just handle the edges in the constructor, and allow any new edges to be duplicated if it's not directed.